From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 21 Apr 2022 02:01:18 +0200
Subject: Osk-layouts: Add Austrian German layout

The 'at' layout is identical to 'de', but as it is implemented as
a different layout rather than a variant, we can only fall back
to 'us'.

To fix that, copy the existing German layout into a new Austrian
one (with the 'name' and 'locale' keys adjusted accordingly).

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5372
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2274>
Origin: upstream, 42.1, commit:908156f4bdaa9688ace92bd9a41c4f79f58cfeca
---
 data/gnome-shell-osk-layouts.gresource.xml |   1 +
 data/osk-layouts/at.json                   | 570 +++++++++++++++++++++++++++++
 2 files changed, 571 insertions(+)
 create mode 100644 data/osk-layouts/at.json

diff --git a/data/gnome-shell-osk-layouts.gresource.xml b/data/gnome-shell-osk-layouts.gresource.xml
index c5f22ef..2b454f3 100644
--- a/data/gnome-shell-osk-layouts.gresource.xml
+++ b/data/gnome-shell-osk-layouts.gresource.xml
@@ -3,6 +3,7 @@
   <gresource prefix="/org/gnome/shell/osk-layouts">
     <file>am.json</file>
     <file>ara.json</file>
+    <file>at.json</file>
     <file>be.json</file>
     <file>bg.json</file>
     <file>by.json</file>
diff --git a/data/osk-layouts/at.json b/data/osk-layouts/at.json
new file mode 100644
index 0000000..6eefa06
--- /dev/null
+++ b/data/osk-layouts/at.json
@@ -0,0 +1,570 @@
+{
+  "levels": [
+    {
+      "level": "",
+      "mode": "default",
+      "rows": [
+        [
+          [
+            "q"
+          ],
+          [
+            "w"
+          ],
+          [
+            "e",
+            "é",
+            "è",
+            "ê",
+            "ë",
+            "ė"
+          ],
+          [
+            "r"
+          ],
+          [
+            "t"
+          ],
+          [
+            "z"
+          ],
+          [
+            "u",
+            "ü",
+            "û",
+            "ù",
+            "ú",
+            "ū"
+          ],
+          [
+            "i"
+          ],
+          [
+            "o",
+            "ö",
+            "ô",
+            "ò",
+            "ó",
+            "õ",
+            "œ",
+            "ø",
+            "ō"
+          ],
+          [
+            "p"
+          ]
+        ],
+        [
+          [
+            "a",
+            "ä",
+            "â",
+            "à",
+            "á",
+            "æ",
+            "ã",
+            "å",
+            "ā"
+          ],
+          [
+            "s",
+            "ß",
+            "ś",
+            "š"
+          ],
+          [
+            "d"
+          ],
+          [
+            "f"
+          ],
+          [
+            "g"
+          ],
+          [
+            "h"
+          ],
+          [
+            "j"
+          ],
+          [
+            "k"
+          ],
+          [
+            "l"
+          ]
+        ],
+        [
+          [
+            "y"
+          ],
+          [
+            "x"
+          ],
+          [
+            "c"
+          ],
+          [
+            "v"
+          ],
+          [
+            "b"
+          ],
+          [
+            "n",
+            "ñ",
+            "ń"
+          ],
+          [
+            "m"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "shift",
+      "mode": "latched",
+      "rows": [
+        [
+          [
+            "Q"
+          ],
+          [
+            "W"
+          ],
+          [
+            "E",
+            "É",
+            "È",
+            "Ê",
+            "Ë",
+            "Ė"
+          ],
+          [
+            "R"
+          ],
+          [
+            "T"
+          ],
+          [
+            "Z"
+          ],
+          [
+            "U",
+            "Ü",
+            "Û",
+            "Ù",
+            "Ú",
+            "Ū"
+          ],
+          [
+            "I"
+          ],
+          [
+            "O",
+            "Ö",
+            "Ô",
+            "Ò",
+            "Ó",
+            "Õ",
+            "Œ",
+            "Ø",
+            "Ō"
+          ],
+          [
+            "P"
+          ]
+        ],
+        [
+          [
+            "A",
+            "Ä",
+            "Â",
+            "À",
+            "Á",
+            "Æ",
+            "Ã",
+            "Å",
+            "Ā"
+          ],
+          [
+            "S",
+            "SS",
+            "Ś",
+            "Š"
+          ],
+          [
+            "D"
+          ],
+          [
+            "F"
+          ],
+          [
+            "G"
+          ],
+          [
+            "H"
+          ],
+          [
+            "J"
+          ],
+          [
+            "K"
+          ],
+          [
+            "L"
+          ]
+        ],
+        [
+          [
+            "Y"
+          ],
+          [
+            "X"
+          ],
+          [
+            "C"
+          ],
+          [
+            "V"
+          ],
+          [
+            "B"
+          ],
+          [
+            "N",
+            "Ñ",
+            "Ń"
+          ],
+          [
+            "M"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "1",
+            "¹",
+            "½",
+            "⅓",
+            "¼",
+            "⅛"
+          ],
+          [
+            "2",
+            "²",
+            "⅔"
+          ],
+          [
+            "3",
+            "³",
+            "¾",
+            "⅜"
+          ],
+          [
+            "4",
+            "⁴"
+          ],
+          [
+            "5",
+            "⅝"
+          ],
+          [
+            "6"
+          ],
+          [
+            "7",
+            "⅞"
+          ],
+          [
+            "8"
+          ],
+          [
+            "9"
+          ],
+          [
+            "0",
+            "ⁿ",
+            "∅"
+          ]
+        ],
+        [
+          [
+            "@"
+          ],
+          [
+            "#"
+          ],
+          [
+            "€",
+            "¢",
+            "£",
+            "$",
+            "¥",
+            "₱"
+          ],
+          [
+            "%",
+            "‰"
+          ],
+          [
+            "&"
+          ],
+          [
+            "-",
+            "_",
+            "–",
+            "—",
+            "·"
+          ],
+          [
+            "+",
+            "±"
+          ],
+          [
+            "(",
+            "<",
+            "{",
+            "["
+          ],
+          [
+            ")",
+            ">",
+            "}",
+            "]"
+          ]
+        ],
+        [
+          [
+            "*",
+            "†",
+            "‡",
+            "★"
+          ],
+          [
+            "\"",
+            "“",
+            "”",
+            "«",
+            "»"
+          ],
+          [
+            "'",
+            "‘",
+            "’",
+            "‹",
+            "›"
+          ],
+          [
+            ":"
+          ],
+          [
+            ";"
+          ],
+          [
+            "!",
+            "¡"
+          ],
+          [
+            "?",
+            "¿"
+          ]
+        ],
+        [
+          [
+            "_"
+          ],
+          [
+            "/"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt+shift",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "~"
+          ],
+          [
+            "`"
+          ],
+          [
+            "|"
+          ],
+          [
+            "•",
+            "♪",
+            "♥",
+            "♠",
+            "♦",
+            "♣"
+          ],
+          [
+            "√"
+          ],
+          [
+            "Π",
+            "π"
+          ],
+          [
+            "÷"
+          ],
+          [
+            "×"
+          ],
+          [
+            "¶",
+            "§"
+          ],
+          [
+            "∆"
+          ]
+        ],
+        [
+          [
+            "£"
+          ],
+          [
+            "¥"
+          ],
+          [
+            "$",
+            "¢"
+          ],
+          [
+            "¢"
+          ],
+          [
+            "^",
+            "↑",
+            "↓",
+            "←",
+            "→"
+          ],
+          [
+            "°",
+            "′",
+            "″"
+          ],
+          [
+            "=",
+            "≠",
+            "≈",
+            "∞"
+          ],
+          [
+            "{"
+          ],
+          [
+            "}"
+          ]
+        ],
+        [
+          [
+            "\\"
+          ],
+          [
+            "©"
+          ],
+          [
+            "®"
+          ],
+          [
+            "™"
+          ],
+          [
+            "℅"
+          ],
+          [
+            "["
+          ],
+          [
+            "]"
+          ]
+        ],
+        [
+          [
+            "<",
+            "‹",
+            "≤",
+            "«"
+          ],
+          [
+            ">",
+            "›",
+            "≥",
+            "»"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    }
+  ],
+  "locale": "de-AT",
+  "name": "Austrian German"
+}
